### PR TITLE
ixfrdist: Add option to limit AXFR record count

### PR DESCRIPTION
**Pull request by pieterlexis**
### Short description
This allows the operator to limit the number of records it is willing to receive to prevent resource exhaustion.

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [x] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: pdns/ixfrdist.cc [hunk 1]
```diff
@@ -316,6 +316,9 @@ void updateThread(const string& workdir, const uint16_t& keep, const uint16_t& a
               soa = getRR<SOARecordContent>(dr);
             }
           }
+          if (axfrMaxRecords != 0 && nrecords > axfrMaxRecords) {
```
Comment (rgacogne) [inline pdns/ixfrdist.cc]: Did you ponder moving this into `AXFRRetriever`, so we can reuse it later for `RPZ` or other regular `AXFR` stuff?

File: pdns/ixfrdist.cc [hunk 2]
```diff
@@ -780,6 +783,16 @@ bool parseAndCheckConfig(const string& configpath, YAML::Node& config) {
     config["keep"] = 20;
   }
 
+  if (config["axfr-max-records"]) {
+    try {
+      config["axfr-max-records"].as<uint16_t>();
```
Comment (rgacogne) [inline pdns/ixfrdist.cc]: Shouldn't this be `uint32_t` instead?

Comment (chbruyand) [inline pdns/ixfrdist.cc]: `uint32_t` ?

**Comment by rgacogne** [comment]
This PR is now conflicted, sorry about that!

File: pdns/ixfrdist.cc (hunk 1 above)
Comment (pieterlexis) [reply pdns/ixfrdist.cc]: Yes, let's move this to a new ticket :)

**Comment by pieterlexis** [comment]
> Can somebody? :)

Done!

```
[INFO] Attempting to retrieve SOA Serial update for 'large.rpz' from '127.0.0.1:5300'
[INFO] Attempting to receive full zonedata for 'large.rpz'
[WARNING] Could not retrieve AXFR for 'large.rpz': Received more than 50 records in AXFR, aborted
```

**Comment by aerique** [comment]
@rgacogne @chbruyand @Habbie could you guys please review this again?  Preferably this week.

**Pull request merged**